crates/subspace-farmer/src/cluster/farmer.rs

```diff
@@ -561,33 +568,34 @@ async fn identify_responder(
 }
 
 async fn send_identify_broadcast(nats_client: &NatsClient, farms_details: &[FarmDetails]) {
-    farms_details
+    let Some(farm_id_string) = farms_details
+        .first()
+        .map(|farm_details| farm_details.farm_id_string.as_str())
+    else {
+        // No farms, nothing to do
+        return;
+    };
+    let details = farms_details
         .iter()
-        .map(|farm_details| async move {
-            if let Err(error) = nats_client
-                .broadcast(
-                    &ClusterFarmerIdentifyFarmBroadcast {
-                        farm_id: farm_details.farm_id,
-                        total_sectors_count: farm_details.total_sectors_count,
-                        fingerprint: blake3_hash_list(&[
-                            &farm_details.farm_id.encode(),
-                            &farm_details.total_sectors_count.to_le_bytes(),
-                        ]),
-                    },
-                    &farm_details.farm_id_string,
-                )
-                .await
-            {
-                warn!(
-                    farm_id = %farm_details.farm_id,
-                    %error,
-                    "Failed to send farmer identify notification"
-                );
-            }
+        .map(|farm_details| ClusterFarmerIdentifyFarmDetail {
+            farm_id: farm_details.farm_id,
+            total_sectors_count: farm_details.total_sectors_count,
+            fingerprint: blake3_hash_list(&[
+                &farm_details.farm_id.encode(),
+                &farm_details.total_sectors_count.to_le_bytes(),
+            ]),
         })
-        .collect::<FuturesUnordered<_>>()
-        .collect::<Vec<_>>()
-        .await;
+        .collect::<Vec<_>>();
+
+    if let Err(error) = nats_client
+        .broadcast(
+            &ClusterFarmerIdentifyFarmBroadcast { details },
+            farm_id_string,
+        )
+        .await
+    {
+        warn!(%farm_id_string, %error, "Failed to send farmer identify notification");
```

Comment:
Optional: We're also reducing the number of logs here, so whatever we do for caches, let's do it for farms too.